### PR TITLE
chore: use knip to remove unused files and deps

### PR DESCRIPTION
**Pull request by domeccleston**
## What does this PR do?

- Adds https://knip.dev/ to the monorepo root as a dependency along with a config file and run script
- Removes files and dependencies flagged as unused by that script, removing 26 unused files, 53 unused dependencies, and 22 unused dependencies.
- This will speed up install and build times of our apps while improving code health and maintainability

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Chore (refactoring code, technical debt, workflow improvements)
- [ ] Enhancement (small improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How should this be tested?

- Run pnpm build from the monorepo root to ensure that all builds pass with removed dependencies

## Checklist

- [x] Filled out the "How to test" section in this PR
- [x] Read [Contributing Guide](./CONTRIBUTING.md)
- [x] Self-reviewed my own code
- [ ] Commented on my code in hard-to-understand areas
- [x] Ran `pnpm build`
- [x] Ran `pnpm fmt`
- [x] Checked for warnings, there are none
- [x] Removed all `console.logs`
- [x] Merged the latest changes from main onto my branch with `git pull origin main`
- [x] My changes don't cause any responsiveness issues

### CODE REVIEW

File: knip.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+      entry: "checkly.config.ts",
+    },
+    "apps/web": {
+      entry: ["contentlayer.config.ts", "lib/trpc/index.ts", "trpc.config.ts"],
+    },
+    "apps/api": {
+      entry: "src/worker.ts",
+      ignore: ["**/*.test.ts", "src/pkg/testutil/*.ts"],
+    },
+    "internal/billing": {
```
Comment (chronark) [inline knip.ts:16]: Could we do a catchall here?
"
```suggestion
    "internal": {
      "./*/src/index.ts"
    }
```

Comment (chronark) [reply knip.ts:16]: no idea if that would work

File: .github/workflows/lint.yaml [hunk 2]
```diff
@@ -3,6 +3,15 @@ on:
   workflow_call:
 
 jobs:
+  check_for_unused:
+    name: CheckUnused
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+
```
Comment (chronark) [inline .github/workflows/lint.yaml:12]: needs install step, see below

**Review by chronark — CHANGES_REQUESTED** [review]

File: knip.ts (hunk 1 above)
Comment (domeccleston) [reply knip.ts:16]: unfortunately not cause we have a couple of packages in there that need to be ignored (like tinybird) and I don't think we can say 'all folders except x' with glob

**Review by chronark — APPROVED** [review]

**Pull request merged**